src/view_neume.cpp

```diff
@@ -93,249 +97,176 @@ void View::DrawNc(DeviceContext *dc, LayerElement *element, Layer *layer, Staff
 
     int yValue = clefYPosition + pitchOffset + octaveOffset;
 
-    // If the nc is supposed to be a virga and currently is being rendered as a punctum
-    // change it to a virga
-    if (nc->HasAttribute("diagonalright", "u") && fontNo == SMUFL_E990_chantPunctum) {
-        fontNo = SMUFL_E996_chantPunctumVirga;
-    }
-
-    DrawSmuflCode(dc, noteX + xOffset, yValue + yOffset, fontNo, staff->m_drawingStaffSize, false, true);
-
-    dc->EndGraphic(element, this);
-}
-
-void View::DrawNeume(DeviceContext *dc, LayerElement *element, Layer *layer, Staff *staff, Measure *measure)
-{
-    assert(dc);
-    assert(layer);
-    assert(staff);
-    assert(measure);
-
-    Neume *neume = dynamic_cast<Neume *>(element);
+    // Determine grouping component is a part of and its position
+    Neume *neume = dynamic_cast<Neume *>(nc->GetParent());

```

Comment:
You should call `nc->GetFirstParent(NEUME)` instead if one day there is editorial markup in-between.